examples/keras_io/vision/metric_learning.py

```diff
@@ -0,0 +1,323 @@
+"""
+Title: Metric learning for image similarity search
+Author: [Mat Kelcey](https://twitter.com/mat_kelcey)
+Date created: 2020/06/05
+Last modified: 2023/10/27
+Description: Example of using similarity metric learning on CIFAR-10 images.
+Accelerator: GPU
+"""
+"""
+## Overview
+
+Metric learning aims to train models that can embed inputs into a high-dimensional space
+such that "similar" inputs, as defined by the training scheme, are located close to each
+other. These models once trained can produce embeddings for downstream systems where such
+similarity is useful; examples include as a ranking signal for search or as a form of
+pretrained embedding model for another supervised problem.
+
+For a more detailed overview of metric learning see:
+
+* [What is metric learning?](http://contrib.scikit-learn.org/metric-learn/introduction.html)
+* ["Using crossentropy for metric learning" tutorial](https://www.youtube.com/watch?v=Jb4Ewl5RzkI)
+"""
+
+"""
+## Setup
+"""
+
```

Comment:
if this example only works with TF backend you might want to add a section about setting the `KERAS_BACKEND` variable
```
import os
os.environ["KERAS_BACKEND"] = "tensorflow"
```

Reply:
It works with jax as well, I'll add a section on setting the backend👍🏻

Reply:
unless we can completely remove `import tensorflow` from our code this would just be TF backend compatible only.

Reply:
Done.